homeassistant/components/srp_energy/sensor.py

```diff
@@ -14,41 +13,44 @@
 from homeassistant.const import ATTR_ATTRIBUTION, ENERGY_KILO_WATT_HOUR
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
 from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
 
 from .const import (
     ATTRIBUTION,
+    CONF_IS_TOU,
     DEFAULT_NAME,
+    DOMAIN,
     ICON,
+    LOGGER,
     MIN_TIME_BETWEEN_UPDATES,
     SENSOR_NAME,
     SENSOR_TYPE,
-    SRP_ENERGY_DOMAIN,
 )
 
-_LOGGER = logging.getLogger(__name__)
-
 
 async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up the SRP Energy Usage sensor."""
     # API object stored here by __init__.py
-    is_time_of_use = False
-    api = hass.data[SRP_ENERGY_DOMAIN]
-    if entry and entry.data:
-        is_time_of_use = entry.data["is_tou"]
+    is_time_of_use: bool = getattr(entry, CONF_IS_TOU, False)
```

Comment:
This looks incorrect. There's no such attribute on the config entry.

Reply:
It's defined during config. `vol.Optional(CONF_IS_TOU, default=False): bool,`

Reply:
No, not on the config entry. The data is stored as `config_entry.data`.

Have you tested this code with this option set to True?